tracex_project/trace_comparator/comparator.py

```diff
@@ -0,0 +1,230 @@
+    ground_truth_activities = ground_truth_df["activity"]
+
+    (
+        mapping_data_to_ground_truth,
+        mapping_ground_truth_to_data,
+    ) = find_activity_mapping(view, pipeline_activities, ground_truth_activities)
+    missing_activities = find_unmapped_activities(
+        ground_truth_activities, mapping_ground_truth_to_data
+    )
+    unexpected_activities = find_unmapped_activities(
+        pipeline_activities, mapping_data_to_ground_truth
+    )
+    wrong_orders = find_wrong_orders(pipeline_activities, mapping_ground_truth_to_data)
+
+    matching_percent_pipeline_to_ground_truth = find_matching_percentage(
+        pipeline_activities, mapping_data_to_ground_truth
+    )
+    matching_percent_ground_truth_to_pipeline = find_matching_percentage(
+        ground_truth_activities, mapping_ground_truth_to_data
+    )
+
+    results_dict = {
+        "mapping_data_to_ground_truth": mapping_data_to_ground_truth,
+        "mapping_ground_truth_to_data": mapping_ground_truth_to_data,
+        "missing_activities": missing_activities,
+        "unexpected_activities": unexpected_activities,
+        "wrong_orders": wrong_orders,
+        "matching_percent_pipeline_to_ground_truth": matching_percent_pipeline_to_ground_truth,
+        "matching_percent_ground_truth_to_pipeline": matching_percent_ground_truth_to_pipeline,
+    }
+
+    return results_dict
+
+
+def find_activity_mapping(view, pipeline_activities, ground_truth_activities):
+    """Find the activity mapping between two dataframes"""
+    total_steps = len(pipeline_activities) + len(ground_truth_activities)
+    half_progress = len(pipeline_activities)
+
+    mapping_data_to_ground_truth = compare_activities(
+        view,
+        0,
+        total_steps,
+        "Mapping Pipeline Activites to Ground Truth Activites",
+        pipeline_activities,
+        ground_truth_activities,
+    )
+    mapping_ground_truth_to_data = compare_activities(
+        view,
+        half_progress,
+        total_steps,
+        "Mapping Ground Truth Activites to Pipeline Activites",
+        ground_truth_activities,
+        pipeline_activities,
+    )
+
+    (
+        mapping_data_to_ground_truth,
+        mapping_ground_truth_to_data,
+    ) = postprocess_mappings(mapping_data_to_ground_truth, mapping_ground_truth_to_data)
+
+    return mapping_data_to_ground_truth, mapping_ground_truth_to_data
+
+
+def compare_activities(
+    view,
+    current_step,
+    total_steps,
+    status,
+    input_activities,
+    comparison_basis_activities,
+):
+    """Compare input activities with comparison basis activities."""
+    mapping_input_to_comparison = []
+    for index, activity in enumerate(input_activities):
+        update_progress(view, current_step, total_steps, status)
+        find_activity(
+            activity,
+            comparison_basis_activities,
+            index,
+            mapping_input_to_comparison,
+        )
+        time.sleep(2)
```

Comment:
Because else we got token Limit reaches, which crashed the whole Process...

We should think about implementing some Form of safety, that we only query GPT, when the token Limits dont conflict...